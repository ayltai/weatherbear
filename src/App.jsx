import { useOnlineStatus, } from '@ayltai/use-online-status';
import { createMuiTheme, CssBaseline, ThemeProvider, } from '@material-ui/core';
import React from 'react';
import { useTranslation, withTranslation, } from 'react-i18next';
import { Route, Switch, } from 'react-router-dom';

import { About, } from './components/pages/About';
import { Locations, } from './components/pages/Locations';
import { Main, } from './components/pages/Main';
import { Settings, } from './components/pages/Settings';
import { Notification, } from './components/views/Notification';
import { Preferences, } from './models/Preferences';
import { Configurations, } from './Configurations';
import './css/weather-icons.min.css';

const Component = () => {
    const isOnline = useOnlineStatus();

    const { t, } = useTranslation();

    return (
        <ThemeProvider theme={createMuiTheme({
            palette : Configurations.createPalette(Preferences.load().isDarkMode),
        })}>
            <CssBaseline />
            <Notification
                type='error'
                message={t('You are currently offline')}
                closeable={false}
                show={!isOnline} />
            <Switch>
                <Route
                    exact
                    path='/'
                    component={Main} />
                <Route
                    exact
                    path='/settings'
                    component={Settings} />
                <Route
                    exact
                    path='/settings/locations'
                    component={Locations} />
                <Route
                    exact
                    path='/about'
                    component={About} />
            </Switch>
        </ThemeProvider>
    );
};

export const App = withTranslation()(Component);
