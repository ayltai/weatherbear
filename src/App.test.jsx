import { mount, } from 'enzyme';
import React from 'react';
import { BrowserRouter, } from 'react-router-dom';
import renderer from 'react-test-renderer';

import { App, } from './App';

window.require = require;

describe('<App />', () => {
    it('renders correctly', () => {
        expect(renderer.create(
            <BrowserRouter>
                <App />
            </BrowserRouter>
        ).toJSON()).toMatchSnapshot();
    });

    it('mounts without error', () => {
        mount(
            <BrowserRouter>
                <App />
            </BrowserRouter>
        );
    });
});
