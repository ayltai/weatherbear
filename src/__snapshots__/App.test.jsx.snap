// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> renders correctly 1`] = `
Array [
  <span
    className="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
    style={
      Object {
        "height": 512,
        "width": 1024,
      }
    }
  />,
  <span
    className="MuiSkeleton-root MuiSkeleton-rect MuiSkeleton-wave"
    style={
      Object {
        "height": 512,
        "width": 1024,
      }
    }
  />,
  <hr
    className="MuiDivider-root"
  />,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-evenly"
  >
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-wave"
        style={
          Object {
            "height": 48,
            "width": 48,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-wave"
        style={
          Object {
            "height": 48,
            "width": 48,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-wave"
        style={
          Object {
            "height": 48,
            "width": 48,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </div>
    <div
      className="MuiGrid-root MuiGrid-item"
    >
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-circle MuiSkeleton-wave"
        style={
          Object {
            "height": 48,
            "width": 48,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
      <span
        className="MuiSkeleton-root MuiSkeleton-text MuiSkeleton-wave"
        style={
          Object {
            "height": undefined,
            "width": undefined,
          }
        }
      />
    </div>
  </div>,
  <hr
    className="MuiDivider-root"
  />,
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-7"
    />
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root makeStyles-container-126 MuiIconButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Settings"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.95 10.78c.03-.25.05-.51.05-.78s-.02-.53-.06-.78l1.69-1.32c.15-.12.19-.34.1-.51l-1.6-2.77c-.1-.18-.31-.24-.49-.18l-1.99.8c-.42-.32-.86-.58-1.35-.78L12 2.34c-.03-.2-.2-.34-.4-.34H8.4c-.2 0-.36.14-.39.34l-.3 2.12c-.49.2-.94.47-1.35.78l-1.99-.8c-.18-.07-.39 0-.49.18l-1.6 2.77c-.1.18-.06.39.1.51l1.69 1.32c-.04.25-.07.52-.07.78s.02.53.06.78L2.37 12.1c-.15.12-.19.34-.1.51l1.6 2.77c.1.18.31.24.49.18l1.99-.8c.42.32.86.58 1.35.78l.3 2.12c.04.2.2.34.4.34h3.2c.2 0 .37-.14.39-.34l.3-2.12c.49-.2.94-.47 1.35-.78l1.99.8c.18.07.39 0 .49-.18l1.6-2.77c.1-.18.06-.39-.1-.51l-1.67-1.32zM10 13c-1.65 0-3-1.35-3-3s1.35-3 3-3 3 1.35 3 3-1.35 3-3 3z"
            transform="scale(1.2, 1.2)"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root makeStyles-container-159 MuiIconButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Photo by  / Unsplash"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M11 7h2v2h-2zm0 4h2v6h-2zm1-9C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <button
      aria-describedby={null}
      className="MuiButtonBase-root MuiIconButton-root makeStyles-container-160 MuiIconButton-sizeSmall"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseOver={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      title="Refresh"
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.65 6.35C16.2 4.9 14.21 4 12 4c-4.42 0-7.99 3.58-7.99 8s3.57 8 7.99 8c3.73 0 6.84-2.55 7.73-6h-2.08c-.82 2.33-3.04 4-5.65 4-3.31 0-6-2.69-6-6s2.69-6 6-6c1.66 0 3.14.69 4.22 1.78L13 11h7V4l-2.35 2.35z"
          />
        </svg>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </button>
  </div>,
]
`;
